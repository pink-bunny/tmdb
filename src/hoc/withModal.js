import React from 'react';

const withModal = (WrappedComponent) => {
  class WrappedComponentShell extends React.Component {
    constructor(props) {
      super(props);
      this.state = {
        modalVisible: false
      };
    }

    showModal = () => {
      this.setState({ modalVisible: true });
    };

    hideModal = () => {
      this.setState({ modalVisible: false });
    };

    render() {
      const { modalVisible } = this.state;
      const modalProps = {
        modalVisible,
        showModal: this.showModal,
        hideModal: this.hideModal
      };

      return <WrappedComponent {...modalProps} {...this.props} />; /* eslint-disable-line */
    }
  }

  return WrappedComponentShell;
};

export default withModal;
