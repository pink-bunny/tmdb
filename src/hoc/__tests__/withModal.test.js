import React from 'react';
import { shallow } from 'enzyme';

import withModal from '../withModal';

describe('withModal HOC', () => {
  const MockModal = () => <div>Mock modal</div>;
  const WrapedModal = withModal(MockModal);
  const wrapper = shallow(<WrapedModal />);
  const instance = wrapper.instance();

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('has method', () => {
    it('showModal() and set modalVisible to true', () => {
      instance.showModal();
      expect(wrapper.state().modalVisible).toBe(true);
    });

    it('hideModal() and set modalVisible to false', () => {
      instance.hideModal();
      expect(wrapper.state().modalVisible).toBe(false);
    });
  });
});
