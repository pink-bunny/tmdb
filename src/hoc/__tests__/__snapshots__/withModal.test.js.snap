// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withModal HOC matches snapshot 1`] = `
<MockModal
  hideModal={[Function]}
  modalVisible={false}
  showModal={[Function]}
/>
`;
